addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/comprehend/ComprehendServiceImpl.java

```diff
@@ -54,12 +65,16 @@ public DetectSentimentResult detectSentiment(String text, String languageCode) {
         }
 
         DetectSentimentRequest request = new DetectSentimentRequest().withText(text).withLanguageCode(languageCode);
-        DetectSentimentResult result = getClient().detectSentiment(request);
 
-        if (log.isDebugEnabled()) {
-            log.debug("DetectSentimentResult is " + result);
+        try {
+            DetectSentimentResult result = getClient().detectSentiment(request);
+            if (log.isDebugEnabled()) {
+                log.debug("DetectSentimentResult is " + result);
+            }
```

Comment:
Why do we need to have an if here ?

Reply:
Same for all others.

Reply:
It's something we're doing in Nuxeo platform too to avoid logging (even if the debug is not activated)